wp-content/plugins/core/src/Templates/Content/Panels/MediaText.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+
+namespace Tribe\Project\Templates\Content\Panels;
+
+use Tribe\Project\Panels\Types\MediaText as MediaTextPanel;
+use Tribe\Project\Templates\Components\Image;
+use Tribe\Project\Templates\Components\Content_Block;
+use Tribe\Project\Templates\Components\Text;
+use Tribe\Project\Templates\Components\Title;
+use Tribe\Project\Templates\Components\Button;
+use Tribe\Project\Templates\Components\Video;
+use Tribe\Project\Theme\Oembed_Filter;
+use Tribe\Project\Theme\Util;
+
+class MediaText extends Panel {
+
+	/**
+	 * Get the data.
+	 *
+	 * @return array
+	 */
+	public function get_data(): array {
+		$data       = parent::get_data();
+		$panel_data = $this->get_mapped_panel_data();
+		$data       = array_merge( $data, $panel_data );
+
+		return $data;
+	}
+
+	/**
+	 * Get the mapped panel data.
+	 *
+	 * @return array
+	 */
+	protected function get_mapped_panel_data(): array {
+		$data = [
+			'content_block'  => $this->get_media_text_content_block(),
+			'media'          => $this->get_media_text_media(),
+			'media_position' => $this->get_media_text_image_position(),
+			'layout'         => $this->get_media_text_layout(),
+		];
+
+		return $data;
+	}
+
+	/**
+	 * Overrides `get_classes()` from the Panel parent class.
+	 *
+	 * Return value is available in the twig template via the `classes` twig variable in the parent class.
+	 *
+	 * @return string
+	 */
+	protected function get_classes(): string {
+		$no_padding = '';
+		$wrapper_class = $this->get_media_text_wrapper_class();
```

Comment:
I actually disagree :)

Any time I find myself doing additional logic inside a function, I find it's always best to break it out into another method so that each one can handle as few things as possible.